eis_toolkit/spatial_analyses/cba.py

```diff
@@ -0,0 +1,511 @@
+    centX = cells_w_bound.centroid.x
+    centY = cells_w_bound.centroid.y
+    df = pd.DataFrame({"CentX": centX, "CentY": centY})
+    points = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.CentX, df.CentY), crs=geodata.crs)
+
+    # Transformation into a disc with a given radius
+    disc = gpd.GeoDataFrame.copy(points)
+    disc["geom"] = disc.geometry.buffer(buff_radius, resolution=16)
+
+    return gpd.GeoDataFrame(disc, geometry="geom", crs=geodata.crs)
+
+
+def prepare_disc(  # type: ignore[no-any-unimported]
+    geodata: gpd.GeoDataFrame,
+    disc: gpd.GeoDataFrame,
+    target_attribut: str,
+    subset_of_target_attribut_values: list,
+) -> Tuple[list, gpd.GeoDataFrame]:
+    """Intermediate utility.
+
+    Compute spatial joint and generate the table of presence of the different
+    attributes.
+
+    Args:
+        geodata: the geodata object corresponding to the vector file.
+        disc: existing disk 'paving'.
+        target_attribut: target attribute to extract from the vector file.
+        subset_of_target_attribut_values: subset of the relevant values
+            of the target attribute
+
+    Returns: pair of variables
+        -> boolean indicating if the target attribute must be dummified
+        -> grid resulting from the intersection between the mesh and the
+            vector file
+    """
+    dummification, target, identified_values = check_and_prepare_param(
+        geodata, target_attribut, subset_of_target_attribut_values
+    )
+
+    # Spatial join of the intersection between the grid and the current map
+    join_grid = gpd.sjoin(disc, geodata, how="inner", predicate="intersects")
+
+    if dummification:
+        tmp = pd.get_dummies(join_grid[join_grid[target].isin(identified_values)][[target]], prefix="", prefix_sep="")
+        indicators = tmp.groupby(tmp.index).max()
+    else:
+        tmp = join_grid[target]
+        indicators = tmp.groupby(tmp.index).max()
+    return indicators, join_grid
+
+
+def get_grid(  # type: ignore[no-any-unimported]
+    working_map: gpd.GeoDataFrame, cell_width: int, cell_height: int
+) -> gpd.GeoDataFrame:
+    """Intermediate utility.
+
+    Produces a regular grid based on geodata surface.
+    Paving begin from the upper left corner (North-West) and go to the South
+    and East (up to down, left to right paving).
+
+    Args:
+        working_map: geodata used as a 'bounding box'.
+        cell_width: width of cells.
+        cell_height: height of cells.
+
+    Returns:
+        GeoDataFrame containing all the cells paving the geodata area.
+    """
+    xmin, ymin, xmax, ymax = working_map.total_bounds
+    rows = int(np.ceil((ymax - ymin) / cell_height))
+    cols = int(np.ceil((xmax - xmin) / cell_width))
+    x_left_origin = xmin
+    x_right_origin = xmin + cell_width
+    y_top_origin = ymax
+    y_bottom_origin = ymax - cell_height
+    print("Columns: %s" % cols)
+    print("Rows: %s" % rows)
+    print("Cells: %s" % (cols * rows))
```

Comment:
You can use `logging` to report runtime info. `print` should be avoided as it cannot be controlled outside the function.

```suggestion
    import logging # at top of the module
   
    logging.info("Columns: %s" % cols)
    logging.info("Rows: %s" % rows)
    logging.info("Cells: %s" % (cols * rows))
```

Or alternatively just remove the `print` code.

Reply:
I chose the latter. Those information should be easy to retrieve from the shapefile produced or even the tiff.